lib/chef/provider/apt_preference.rb

```diff
@@ -0,0 +1,106 @@
+#
+# Author:: Tim Smith (<tsmith@chef.io>)
+# Copyright:: Copyright (c) 2016 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/resource"
+require "chef/dsl/declare_resource"
+require "chef/mixin/which"
+require "chef/provider/noop"
+require "chef/log"
+
+class Chef
+  class Provider
+    class AptPreference < Chef::Provider
+      use_inline_resources
+
+      extend Chef::Mixin::Which
+
+      provides :apt_preference do
+        which("apt-get")
+      end
```

Comment:
this is probably going to do the same thing where it binds super tightly and overrides the cookbook.